api/src/main/java/org/pmiops/workbench/exceptions/ExceptionAdvice.java

```diff
@@ -9,22 +9,32 @@
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.servlet.NoHandlerFoundException;
 
 @ControllerAdvice
 public class ExceptionAdvice {
   private static final Logger log = Logger.getLogger(ExceptionAdvice.class.getName());
 
   @ExceptionHandler({HttpMessageNotReadableException.class})
-  public ResponseEntity<?> messageNotReadableError(Exception e) {
+  public ResponseEntity<ErrorResponse> messageNotReadableError(Exception e) {
     log.log(Level.INFO, "failed to parse HTTP request message, returning 400", e);
     return ResponseEntity.status(HttpStatus.BAD_REQUEST)
         .body(
             WorkbenchException.errorResponse("failed to parse valid JSON request message")
                 .statusCode(HttpStatus.BAD_REQUEST.value()));
   }
 
+  @ExceptionHandler({NoHandlerFoundException.class})
```

Comment:
I found some docs indicating that "not found" handling changed in recent versions of Spring, so I expect that the upgrade is what caused the problem